src/background/lotus-client/Lotus.js

```diff
@@ -214,14 +257,25 @@ class Lotus {
     //
     // Generate the PCH create message
     //
-    var signedCreateMessage; // TODO:  does this need to be declared out here?
+    var signedCreateMessage;
     try {
-      let create_pymtchan = signer.createPymtChanWithFee(fromAddr, toAddr, `${amountAttoFil}`, nonce, "10000000", "16251176117", "140625002"); // gas limit, fee cap, premium
-      ports.postLog("DEBUG: Lotus.createPaymentChannel: create_pymtchan="+inspect(create_pymtchan))
-      // TODO:  use gas esimator:
-      //create_pymtchan = await filRPC.getGasEstimation(create_pymtchan)
+      let gasEstimation = await this.getGasEstimation({
+        "From": fromAddr,
+        "To": toAddr,
+        "Nonce": nonce,
+        "Value": `${amountAttoFil}`,
```

Comment:
This message needs to match the createPymtChanWithFee message as much as possible.
In this case, you are missing the Params and Method.

Reply:
Ah ok, thanks!  

ping @aistoc

Reply:
The `to` is also invalid, as the message for creation of the Payment Channel will be sent to init actor, not to the payment channel recipient.
In essence, what Gas Estimation does is it executes the message and takes a look at it gas used, it cannot do that if inputs are for different kind of message or different receipent.